src/aws-manager.js

```diff
@@ -24,6 +24,102 @@ function dateForInflux(thingy) {
 }
 
 /**
+ * This method will run an EC2 operation.  Because the AWS-SDK client is so
+ * much fun to work with, we need to do the following things above what it does
+ * to get useful information out of it.
+ *
+ *   1. We want to have exceptions that always have region, method and service
+ *      name if available
+ *   2. Any requests which would have a requestId should include it in their
+ *      exceptions
+ *   3. Sometimes promises from AWS-SDK just magically never return and never
+ *      timeout even though we've set those options.  We have our own timeout
```

Comment:
yes.  The `aws-sdk` is just not that good in my opinion, this is an example.  This is why I lean more towards using `aws4`; i don't trust `aws-sdk` to always actually work.